lib/rms_norm.cpp

```diff
@@ -0,0 +1,68 @@
+#include "flag_gems/operators.h"
+#include "flag_gems/utils.h"
+
+#include <iostream>
+#include "c10/cuda/CUDAStream.h"
+#include "triton_jit/triton_jit_function.h"
+
+namespace flag_gems {
+using namespace triton_jit;
+
+at::Tensor rms_norm(const at::Tensor& input,   // [..., hidden_size]
+                    const at::Tensor& weight,  // [hidden_size]
+                    double epsilon) {          //  default 1e-5
+
+  int64_t hidden_size = input.size(-1);
+  int64_t M = input.numel() / hidden_size;
+  int64_t N = weight.size(0);  // assumes 1D weight
+  int64_t BLOCK_SIZE = utils::next_power_of_2(N);
+
+  at::Tensor out = at::empty(input.sizes(), input.options());
+  at::Tensor inv_rms = at::empty({M}, at::TensorOptions().dtype(torch::kFloat32).device(input.device()));
+
+  auto input_strides = input.strides();
+  auto output_strides = out.strides();
+
+  const TritonJITFunction& f =
+      TritonJITFunction::getInstance(std::string(utils::get_triton_src_path() / "rms_norm.py"),
+                                     "rms_norm_kernel");
+
+  // getCurrentCUDAStream ensures that the stream is initialized, a default stream for each device
+  c10::cuda::CUDAStream stream = c10::cuda::getCurrentCUDAStream();
+  c10::DeviceGuard guard(out.device());
```

Comment:
It is better to apply device guard first. We can swap the 2 lines in the future.